lib/booking_central.rb

```diff
@@ -0,0 +1,92 @@
+#require_relative 'room'
+require_relative 'date_range'
+require_relative 'reservation'
+require_relative 'block'
+
+class BookingCentral 
+  attr_accessor :rooms, :all_reservations, :blocks
+  
+  def initialize
+    @blocks = []
+    @rooms = []
+    @all_reservations = []
+
+    (1..20).each do |number|
+      room = Room.new(number, room_reservations: [])
+      @rooms << room
+    end
+  end
+
+  def reservations_by_date(check_in, check_out)
+    asking_date = DateRange.generate_date_range(check_in, check_out)
+    matching_reservations = @all_reservations.select { 
+      |reservation, details| reservation if DateRange.dates_overlap?(reservation.date_range, asking_date)
+    }
+    return matching_reservations
+  end
+
+  def list_available_rooms(check_in, check_out)
+    asking_date = DateRange.generate_date_range(check_in, check_out)
+    available_rooms = @rooms - (reservations_by_date(check_in, check_out)).map{ |reservation| reservation.room }
+    available_rooms -= @rooms.select{ |room| room.blocked == true }
```

Comment:
First, I love that you've broken this logic out as a separate method, that makes the rest of your code much easier to read.

Second, good work getting this tricky logic with the lists of rooms sorted out.

Third, this logic is complex enough (particularly line 30) that it would probably make sense to split it across multiple lines.